src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -64,4 +64,20 @@ public int hashCode() {
     hash = 31 * hash + (wikiLink == null ? 0 : Objects.hashCode(wikiLink));
     return hash;
   }
+
+  public String getName() {
+    return name;
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  /** 
+   * Method to add a new type to the existing entity, removing the need of duplicate
+   * entities which have the same name but different types.
+   */
+  public void addType(String newType) {
+    this.type += ", " + newType;
```

Comment:
Change type from String to List of Strings. In this case you don't need to crate splitter and split then.